core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -187,8 +189,8 @@ public int getOutgoingRecoveries(String nodeId) {
         return recoveriesPerNode.getOrDefault(nodeId, Recoveries.EMPTY).getOutgoing();
     }
 
-    private ShardRouting findPrimary(ShardRouting routing) {
-        List<ShardRouting> shardRoutings = assignedShards.get(routing.shardId());
+    public ShardRouting findAssignedPrimary(ShardId shardId) {
```

Comment:
right, I used it in ThrottlingAllocationDecider, then saw that we have the nicer method "activePrimary" here.